core/src/app.module.ts

```diff
@@ -42,8 +100,17 @@ import { HealthModule } from './health/health.module';
           return enforcer;
         },
       },
-      usernameFromContext: (ctx) =>
-        ctx.switchToHttp().getRequest<Request>().user.username,
+      userFromContext: (ctx) => {
+        const req = ctx.getArgs()[2]['request'];
+        const user = getUserFromRequest(req);
+
+        if (user) {
+          Logger.log(`Found user from request: ${JSON.stringify(user)}.`);
+          return user.id;
+        } else {
+          return 'anonymous';
+        }
```

Comment:
Doesn't this need to return a full user? a la https://github.com/SprocketBot/sprocket/blob/v2/lib/types/auth/user.ts#L11-L16

Reply:
It looks like both ways are supported, as long as the returned object has a 'username' field. https://github.com/node-casbin/nest-authz/blob/master/README.md